### PR TITLE
AUI-3207 - Add some steps detailing how to do a release

### PR DESCRIPTION
**Pull request by javiergamarra**
Trying to explain the release process... feedback is welcomed :)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -111,4 +111,53 @@ When you have successfully run the command to put this local version on your mac
...
+
+* `npm install` (if you have not done it already)
+* `gulp build` (if you have not done it already)
+* `gulp maven-publish-snapshot` 
+
+This should create a jar under this [folder](https://repository.liferay.com/nexus/content/repositories/liferay-public-snapshots/com/liferay/webjars/com.liferay.webjars.alloy-ui/). The size of the binary should be around 25MB. You can publish as many times as you want under that folder (and it will rename the binary to have a numeric suffix like -2.jar).
+
+### Common issues
+
+If the size is way smaller than 25MB it could mean that you have missed a step (most probably `gulp build`). If the size is way bigger than 25MB it is probably because the publish script is picking local files under the `alloy-ui` folder. Remove binary files/jar generated by `maven install` or move the `.git` folder outside the main folder while you are generating.
```
Comment (markocikos) [inline CONTRIBUTING.md:155]: Moving `.git` folder? :scream: If the script that builds `jar` grabs the `.git` folder, that sounds like a bug to me. 

`maven install` I can understand, if we expect `maven` to be installed globally. If that's the case, we should probably commit it to the repo. Or, at the very least, document global install as one of the release steps.

**Review by markocikos — COMMENTED** [review]
@nhpatt left a few comments inline.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -111,4 +111,53 @@ When you have successfully run the command to put this local version on your mac
...
+
+### Common issues
+
+If the size is way smaller than 25MB it could mean that you have missed a step (most probably `gulp build`). If the size is way bigger than 25MB it is probably because the publish script is picking local files under the `alloy-ui` folder. Remove binary files/jar generated by `maven install` or move the `.git` folder outside the main folder while you are generating. 
+
+If you receive a 401 it's probable that maven is not picking your credentials correctly, double check the paths and see if maven is using your custom settings.xml. If you receive a 403, it means that your credentials are not valid or you don't have access to publish in that folder.
+
+### Release in final repository
+
+When you are satisfied with the generated .jar, you can launch `gulp maven-publish-snapshot` to publish the jar in the release repository. This version can not be updated/replaced.
```
[the remark this answers is not in the document]
Comment (nhpatt) [reply CONTRIBUTING.md:161]: Bad redaction, L151 says This should create a jar under this [folder] => link to remote folder in nexus

[the remark this answers is not in the document]
Comment (markocikos) [reply CONTRIBUTING.md:161]: Oh I see, I misunderstood that is was creating a jar in local dir. But then, what's the difference between 'publishing a test version' and 'releasing in final repository'? Based on docs, they are done with the same command, `gulp maven-publish-snapshot`

**Pull request closed without merging**